src/base/bittorrent/session.cpp

```diff
@@ -4016,7 +4055,10 @@ void Session::handleTorrentDeletedAlert(const lt::torrent_deleted_alert *p)
     if (!m_removingTorrents.contains(infoHash))
         return;
     const RemovingTorrentData tmpRemovingTorrentData = m_removingTorrents.take(infoHash);
-    Utils::Fs::smartRemoveEmptyFolderTree(tmpRemovingTorrentData.savePathToRemove);
+    if (tmpRemovingTorrentData.deleteOption == TorrentAndFiles)
+        Utils::Fs::smartRemoveEmptyFolderTree(tmpRemovingTorrentData.savePathToRemove);
+    else if (tmpRemovingTorrentData.deleteOption == TorrentAndFolder)
+        Utils::Fs::removeDirRecursive(tmpRemovingTorrentData.savePathToRemove);
```

Comment:
Will this remove folders outside the torrent? For example, a single-file torrent downloaded (without creating sub-folder) to `/my/path/torrent_file` and will this line remove `/my/path`? I hope it won't.

Reply:
No it won't. It was tested in the original PR and i tested it again to be sure.

Reply:
OK, I had another concerns.
First one. If some torrent has the following structure: `b/c/d/e.txt` and is saved under `/a`. What is the expected behavior under this option? Removing folder `/a/b/c/d`? or removing `/a/b`? I suppose it is the latter, right?

Second. I suspect it is not correct to use `Utils::Fs::removeDirRecursive` to just remove one directory. It should instead remove the top level sub-folders one by one. For example a torrent has `a/b/c/z1.txt`, `a/d/e/z2.txt`, `z3.txt`. Then it should remove `a/b` and `a/d` folders.

Reply:
>Removing folder /a/b/c/d? or removing /a/b? I suppose it is the latter, right?

Yes. As for the rest, what i do know is that this stuff is out of my league and if it's required to do anything other than use `Utils::Fs::removeDirRecursive` then i'll have to abandon ship.